Nemo11235/CS160-Final-Project
# multiplayer-pop-up and singleplayer-pop-up

**Description:**
Upon completion of a multiplayer game, both players will receive a pop-up message declaring the game is over. There are three possible scenarios: First player wins, Second player wins, or there is a Draw. Upon receiving the screen, the player is presented with the correct word, and the player who won the game or a draw message. There are two buttons available for the user to choose from: 'Play Again" or a "Singleplayer" button. The background is blurred and darkened out, so the user has only two choices.
In addition to the multiplayer-pop-up, this branch includes the latest version of singleplayer-pop-up. When the user wins a round, a pop-up will instantly appear to display a party popper image, congratulations messages, and an option to "Play Again" or to close the pop-up at the top left. When the user loses a round, the pop-up message will display "Round Over", and inform the user with the correct word. The current 'Play Again' message will refresh the page.


**Testing for multiplayer-pop-up:**
1. Start react application and server, and open two different tabs for two users
2. Navigate to multiplayer mode for both open tabs (http://localhost:3000/race)
3. Have both players join a server with the same Room ID
4. Let the first player correctly guess the test word 'APPLE', and the pop-up should immediately appear for both players, announcing that the first player has won, along with the correct word and two buttons to 'Play Again' and for 'Singleplayer'
5. Press the 'Singleplayer' button to return to the home page
6. Have both players join a new server with the same Room ID
7. Let the second player correctly guess the word to win, and the pop-up should display immediate for both players
8. Refresh the page for both players to return to the multiplayer lobby
9. Have both players join a new server with the same Room ID
10. Use up all attempts for the first player, and test that nothing happens
11. Use up all attempts for the second player, and test that a pop-up message appears with the message: "There is a draw!"
12. Refresh the page for both players to return to the multiplayer lobby
13. Open the hamburger menu, and see that the menu looks as such, with the 'Dark Theme' and "Custom Words" options removed:
![Hamburger Menu](https://user-images.githubusercontent.com/60405574/166848222-0857b2c4-f3c3-4a53-8802-3bb7eed04bb2.png)
14. Navigate to 'Tutorial', then 'Multiplayer', then 'Singleplayer', then 'About Us'
15. Navigate to User Profile page and check that all elements are in the correct position


**Testing for singleplayer-pop-up:**
1. Start the react application
2. Use all 6 attempts incorrectly, and test that a pop-up instantly appears with the message "Round Over!", displays the correct word, and has an option to 'Play Again'
3. Click the 'Play Again' button, and see that the page shows a new Singleplayer board
4. Use all 6 attempts incorrectly again, and test the pop-up instantly appears, and click the 'X' button at the top-left of the pop-up to test if the pop-up closes correctly
5. Refresh the page to clear the Singleplayer board
6. Type in the word "APPLE" and see that a pop-up appears with a party popper image, congratulations message, "You got the word!" message, and a 'Play Again" button.
7. Click the "Play Again" button to refresh the page and see a new Singplayer board
8. Use up 5 attempts in the new game, and correctly guess the word "Apple" on the 6th attempt, and test if the correct pop-up window appears
